### ISSUE TITLE
Easy way to spoof header - differences in output between Email::Address:XS and Email::Address

### ISSUE THREAD

**Issue by justingit** [pali/Email-Address-XS#5]
I'm a little worried about this, 

```
#!/usr/bin/perl 
use Email::Address::XS 
print "VERSION: " . $Email::Address::XS::VERSION . "\n"; 

my $header = 'test@example.com <spoofer.address@malicious-site.com>'; 

print Email::Address::XS->parse($header)->is_valid() . "\n"; 

for(Email::Address::XS->parse($header)){ 
	print $_->address . "\n"; 
}
```

For me, this returns, 

```
VERSION: 1.04
1
test@example.com
```

I would have expected, "test@example.com" printed to be, "spoofer.address@malicious-site.com". Am I incorrect in this assumption? 

The following, 
```
#!/usr/bin/perl 
use Email::Address; 
print "VERSION: " . $Email::Address::VERSION . "\n"; 

my $header = 'test@example.com <spoofer.address@malicious-site.com>'; 

#print Email::Address->parse($header)->is_valid() . "\n"; 

for(Email::Address->parse($header)){ 
	print $_->address . "\n"; 
}
```

prints, 

```
VERSION: 1.912
spoofer.address@malicious-site.com
```

Seeing the output is different between the two modules, which one is correct?

**Comment by pali** [comment]
First thing which is needed to know is that `->parse` class method expects RFC5322 `mailbox-list` and in perl list context this class method returns list of Email::Address::XS objects where each object represents one RFC5322 `mailbox`.

Now look at your input:
```
test@example.com <spoofer.address@malicious-site.com>
```

This is invalid RFC5322 `mailbox-list` because `@` character is not allowed in unquoted version of RFC5322 `phrase`. So the input is invalid and Email::Address::XS object should **not** return true by `->is_valid` method.

Email::Address::XS tries to be smart and try to parse even _garbage_ on the input and returns something to user. It parsers from left to right and drop trailing output. In this case `test@example.com` is parsed as a valid RFC5322 `addr-spec` from which is created object and the rest of the string (` <spoofer.address@malicious-site.com>`) is dropped.

So with this behavior, it is expected that `->address` method returns `test@example.com`.

I can admit that parser could be modified and extended to try to interpret unquoted `@` character as a valid `phrase` and change the way how the rest of the input string is parsed. But... parsing invalid input strings is hard and I would say the correct behavior is: garbage in, garbage out.

But the issue here is that `->is_valid` returns true, it must return false for this input. I will look at it and try to fix it. Thanks for reporting this issue.

**Comment by justingit** [comment]
I really appreciate this thorough reply - I had a hunch something subtle was happening. 

> But the issue here is that ->is_valid returns true, it must return false for this input. I will look at it and try to fix it. Thanks for reporting this issue.

That would be very helpful. Thank you for taking a look at this,

**Comment by pali** [comment]
Please check the master branch. It should be fixed it there.

**Comment by justingit** [comment]
Thanks! With your recent changes, this prints, "false". 

```
#!/usr/bin/perl 
use Email::Address::XS;
use v5.10;

my $header = 'test@example.com <spoofer.address@malicious-site.com>'; 

Email::Address::XS->parse($header)->is_valid() ? say 'true' : say 'false';
```

**Comment by pali** [comment]
I have released a new version 1.05 which is fixing this issue.
